cse/src/main/java/com/adyen/checkout/cse/ClientSideEncrypter.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2021 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by caiof on 14/1/2021.
+ */
+package com.adyen.checkout.cse
+
+import android.util.Base64
+import com.adyen.checkout.cse.exception.EncryptionException
+import java.math.BigInteger
+import java.security.InvalidAlgorithmParameterException
+import java.security.InvalidKeyException
+import java.security.KeyFactory
+import java.security.NoSuchAlgorithmException
+import java.security.PublicKey
+import java.security.SecureRandom
+import java.security.spec.InvalidKeySpecException
+import java.security.spec.RSAPublicKeySpec
+import java.util.Locale
+import javax.crypto.BadPaddingException
+import javax.crypto.Cipher
+import javax.crypto.IllegalBlockSizeException
+import javax.crypto.KeyGenerator
+import javax.crypto.NoSuchPaddingException
+import javax.crypto.SecretKey
+import javax.crypto.spec.IvParameterSpec
+import kotlin.text.Charsets.UTF_8
+
+/**
+ * Created by andrei on 8/8/16.
+ */
+class ClientSideEncrypter(publicKeyString: String) {
+    private var aesCipher: Cipher? = null
+    private var rsaCipher: Cipher? = null
+    private val secureRandom: SecureRandom
+
+    @Throws(EncryptionException::class)
+    @Suppress("ThrowsCount")
+    fun encrypt(plainText: String): String {
+        val aesKey = generateAesKey()
+        val iv = generateIV()
+        val encrypted: ByteArray = try {
+            aesCipher!!.init(Cipher.ENCRYPT_MODE, aesKey, IvParameterSpec(iv))
```

Comment:
Can we get rid of the `!!`? I can see a bunch of them around the project so feel free to fix them all (the fixable ones at least) in another PR 😄

Reply:
I've been trying to get rid of most of them, but maybe I'm not being very thorough. I'll check this one out, thx!